src/js/models/panel_model.js

```diff
@@ -390,10 +390,18 @@
                 }
                 else {
                     var theZ = this.get('theZ');
+                    var deltaZ = theZ;
+                    
+                    if (ref_idx) {
+                        var shift = parseInt(ref_idx)
+                        if(!isNaN(shift)){
+                            deltaZ = theZ + shift;
+                        }
+                    }
                     if (format === "pixel") {
                         text = "" + (theZ + 1);
```

Comment:
I wonder if you could also use `deltaZ` instead of `theZ` here, so that you can use `offset` when showing T index.
And if you could apply the same change to `theT` that would be great.
There's maybe not much use case for that (I only noticed it in testing) but that's still the way you'd expect it to behave.
In my test, the white labels (offset=2) show the same as yellow labels (no offset):

![Screenshot 2023-10-18 at 09 54 31](https://github.com/ome/omero-figure/assets/900055/23543135-95fa-4427-ab8d-f3020b503cfe)

Reply:
Ok make sense.
Is it what you had in mind ?

Reply:
Sorry, no! I just meant for that line to be ` text = "" + (deltaZ + 1);` instead of ` text = "" + (theZ + 1);`
That's all! Apologies for the confusion!

Reply:
Ah...Ok, now I see.
I've corrected it. I also correct a small bug on time offset (see below) but I noticed that the current implementation doesn't allow for negative offsets for time (but allowed for Z)
```
                var shift = this.get('deltaT')[shiftIdx];
                deltaT = shift==null ? deltaT : shift;
```

Do we also allow it for time ?